enterprise/pipeline.py

```diff
@@ -0,0 +1,24 @@
+from openedx_filters import PipelineStep
+
+
+class InviteOnlyEnrollments(PipelineStep):
+    """
+    Add the following to the end of edx-platform/lms/envs/devstack.py file:
+
+        OPEN_EDX_FILTERS_CONFIG = {
+            "org.openedx.learning.course.enrollment.started.v1": {
+                "fail_silently": False,
+                "pipeline": [
+                    "enterprise.pipeline.InviteOnlyEnrollments"
+                ]
+            }
+        }
+
+    This will hook into the CourseEnrollmentStarted filter here: https://github.com/openedx/edx-platform/blob/592bc66b3fe5c14bafb7e44ea9e57328fe1dde06/common/djangoapps/student/models/course_enrollment.py#L658-L660
+    and allow you to inspect the data that is being passed to the filter.
+    """
+
+    def run_filter(self, *args, **kwargs):
+        import ipdb
+        ipdb.set_trace()
+        return {}
```

Comment:
From the ticket description:

>The filter hook will grab the relevant information from the request object like the enterprise id, catalog id, user id..etc., validate that they are allowed to enroll in the course and create the CourseEnrollmentAllowed object. This will allow them to successfully enroll without staff having to explicitly enrolle each user.

Unless we were going to modify the filter itself, it looks like the request object can't be accessed here.

The signature of this method is the same as [this one](https://github.com/openedx/openedx-filters/blob/1aab44b4685e8c41553115e1f7d22059e089d69a/openedx_filters/learning/filters.py#L154).